src/main.rs

```diff
@@ -37,6 +37,100 @@ use std::{
 
 thread_local! { static IS_PERF: RefCell<bool> = RefCell::new(false) }
 
+// Inspired by fish's acquire_tty_or_exit
+#[cfg(unix)]
+fn take_control(interactive: bool) {
+    use nix::{
+        errno::Errno,
+        sys::signal::{self, SaFlags, SigAction, SigHandler, SigSet, Signal},
+        unistd::{self, Pid},
+    };
+
+    let shell_pgid = unistd::getpgrp();
+    let owner_pgid = unistd::tcgetpgrp(nix::libc::STDIN_FILENO).expect("tcgetpgrp");
```

Comment:
I am running into a new problem with this change:

Trying to launch interactive nushell via the [CodeLLDB VS Code extension](https://marketplace.visualstudio.com/items?itemName=vadimcn.vscode-lldb) causes a panic on this line.

![grafik](https://user-images.githubusercontent.com/15833959/194414987-5b78a658-c3ca-47da-8ecc-bf80cc066d1f.png)

It appears that fd `0` is a valid tty and the `atty` crate is reporting the same thing. I don't fully grasp how we can deal with the other error cases vaguely described in https://man7.org/linux/man-pages/man3/tcgetpgrp.3.html

Reproducibility notes: both on Ubuntu standalone and with WSL2 I was able to cause this crash (for the external and internal terminal option in the `launch.json` config).

Launching `gdb target/debug/nu` was not picking up this panic if I get this correctly (but my gdb foo is not strong enough to properly launch nu in a separate terminal while being attached from the start.) Regular late attaching with `gdb -p <nu pid>` was not causing unexpected problems.

Reply:
hm, does fish work in the same place?

Reply:
Will have to read up on the launch configuration options for that.

Reply:
I opened a separate issue for that problem, even if fish were to crash there we shouldn't to enable us to debug other things on startup.